repair/row_level.cc

```diff
@@ -650,18 +653,59 @@ future<std::list<repair_row>> to_repair_rows_list(repair_rows_on_wire rows, sche
     co_return std::move(row_list);
 }
 
+std::unordered_set<dht::token> get_tokens_from_repair_row_list(const std::list<repair_row>& rows) {
+    std::unordered_set<dht::token> ret;
+    for (auto& row : rows) {
+        // Get the tokens for the rows that are read from the node itself and
+        // ignore the rows received from peer nodes. A row with dirty_on_master
+        // flag set comes from peer node. See to_repair_rows_list.
+        if (!row.dirty_on_master()) {
+            ret.insert(row.get_dk_with_hash()->dk.token());
+        }
+    }
```

Comment:
This can't be very large because the number of rows each repair round works on is bounded. So the memory is bounded too. 

This might cause stalls. I am aware of this. I haven't futurized it yet to make the change less noise.